apps/aecore/src/aec_consensus_hc.erl

```diff
@@ -347,19 +393,64 @@ cache_child_epoch_info(Epoch, Height, StartTime) ->
             ok
     end.
 
-get_child_epoch_info(Epoch) ->
+%% Cache stores a tuple {MaxEpoch, [Info...]} where Info :: {Epoch, Height, StartTime}
+%% If the epoch cache is empty and start time is unknown, i will
+%% * Try read epoch first block, if it can, and its time will be epoch start
+%% * Else, will try read last block of previous epoch, then its time + block production time
+%%   will be the epoch start
```

Comment:
Isn't it block_time that decides the time between blocks, not block_production_time?

Reply:
Makes sense, changed. Thanks for catching!
Not sure how to test this, and the problem of not closing the previous epoch is an unsolved problem, the chain will halt if this happens. Otherwise the value is always cached in ETS and this code normally does not run, only for an emergency situation.